### ISSUE TITLE
AppConfig: consolidate testing of single_dataset to one place

### ISSUE THREAD

**Issue by bmccandless** [chanzuckerberg/cellxgene#1883]
If the server is running in single_dataset mode, it runs through a series of checks on that file:

- checks the size (so it can report a message if large)
- checks the about link
- attempts to validate user annotations in some cases
- a couple checks for reembeddings
- checks for diffexp (to give a message if we think the dataset is large and DE will take a long time)

The checks are dispersed in different functions.

It may be better code organization to do all those checks in one place, 
and after the rest of the config has been validated.

**Comment by signechambers1** [comment]
@maniarathi is this a priority or ok to close?

**Comment by maniarathi** [comment]
Yep -- I think if we revisited this, we'd redo most of the config mechanism anyways.
